staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -493,6 +493,23 @@ func (r *crdHandler) getOrCreateServingInfoFor(crd *apiextensions.CustomResource
 	statusScopes := map[string]*handlers.RequestScope{}
 	scaleScopes := map[string]*handlers.RequestScope{}
 
+	structuralSchemas := map[string]*structuralschema.Structural{}
+	for _, v := range crd.Spec.Versions {
+		val, err := apiextensions.GetSchemaForVersion(crd, v.Name)
+		if err != nil {
+			utilruntime.HandleError(err)
+			return nil, fmt.Errorf("the server could not properly serve the CR schema")
+		}
+		if val == nil {
+			continue
+		}
+		structuralSchemas[v.Name], err = structuralschema.NewStructural(val.OpenAPIV3Schema)
+		if *crd.Spec.PreserveUnknownFields == false && err != nil {
+			utilruntime.HandleError(err)
+			return nil, err // validation should avoid this
```

Comment:
the `utilruntime.HandleError(err)` line is for the admin. the error returned here will be shown to the custom resource API caller... do we want to expose the structural schema error details to them or just say something like "an error occurred constructing the type schema", and defer to the condition on the CRD for investigation?

Reply:
done